src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -49,128 +45,141 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
 
         private async Task ExecuteEndpointAsync(string path, HttpContext context, EndPoint endpoint)
         {
-            // Server sent events transport
-            if (context.Request.Path.StartsWithSegments(path + "/sse"))
```

Comment:
I think FXCop would complain about the cyclomatic complexity of this method... I honestly don't know if this refactoring makes it better or worse yet...